server/backend/users/views.py

```
@@ -130,6 +127,68 @@ def post(self, request):
+            uid = urlsafe_base64_decode(uidb64).decode()
+            user = User.objects.get(pk=uid)
+
+            if not PasswordResetTokenGenerator().check_token(user, token):
+            if not PasswordResetTokenGenerator().check_token(user, token):
+                return Response({"error": "Invalid or expired token"}, status=status.HTTP_400_BAD_REQUEST)
+
+            # Validate password complexity
+            if len(password) < 8:
+                return Response(
+                    {"error": "Password must be at least 8 characters long"},
+                    status=status.HTTP_400_BAD_REQUEST
```

Ruff (0.8.2)

Trailing comma missing

(COM812)